### PR TITLE
Move test generated clients into a buildable crate

### PR DESCRIPTION
**Pull request by jayvdb**
This only moves them into a crate to allow building them.
The next step would be to add a CI job that builds them.

### CODE REVIEW

File: progenitor-impl/tests/output/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+pub mod buildomat_builder;
+pub mod buildomat_builder_tagged;
+pub mod buildomat_cli;
+//pub mod buildomat_httpmock;
```
Comment (jayvdb) [inline progenitor-impl/tests/output/src/lib.rs:4]: commented out due to https://github.com/oxidecomputer/progenitor/issues/637

**Comment by ahl** [comment]
What would you think about doing the `JsonSchema` -> `schemars::JsonSchema` and the file renames first?

**Comment by jayvdb** [comment]
There is a lot of overlap, so would be good to get https://github.com/oxidecomputer/progenitor/pull/641 in quickly.

**Comment by jayvdb** [comment]
This PR now sits on top of https://github.com/oxidecomputer/progenitor/pull/642

**Comment by jayvdb** [comment]
@ahl , PR refreshed & green.

**Pull request merged**